src/V3Width.cpp

```diff
@@ -7329,21 +7329,25 @@ class WidthVisitor final : public VNVisitor {
         }
         return false;
     }
-    void checkClassAssign(AstNode* nodep, const char* side, AstNode* rhsp,
+    void checkClassAssign(const AstNode* nodep, const char* side, AstNode* rhsp,
                           AstNodeDType* const lhsDTypep) {
-        if (AstClassRefDType* const lhsClassRefp = VN_CAST(lhsDTypep->skipRefp(), ClassRefDType)) {
-            UASSERT_OBJ(rhsp->dtypep(), rhsp, "Node has no type");
-            AstNodeDType* const rhsDtypep = rhsp->dtypep()->skipRefp();
-            if (AstClassRefDType* const rhsClassRefp = VN_CAST(rhsDtypep, ClassRefDType)) {
+        UASSERT_OBJ(rhsp->dtypep(), rhsp, "Node has no type");
+        const AstNodeDType* const lhsRawDTypep = lhsDTypep->skipRefp();
+        const AstNodeDType* const rhsRawDTypep = rhsp->dtypep()->skipRefp();
+        if (const AstClassRefDType* const lhsClassRefp = VN_CAST(lhsRawDTypep, ClassRefDType)) {
+            if (const AstClassRefDType* const rhsClassRefp
+                = VN_CAST(rhsRawDTypep, ClassRefDType)) {
                 if (isBaseClassRecurse(lhsClassRefp->classp(), rhsClassRefp->classp())) return;
             } else if (rhsp->isNull()) {
                 return;
             }
             nodep->v3error(side << " expects a " << lhsClassRefp->prettyTypeName() << ", got "
-                                << rhsDtypep->prettyTypeName());
+                                << rhsRawDTypep->prettyTypeName());
+        } else if (VN_IS(rhsRawDTypep, ClassRefDType)) {
+            nodep->v3error(side << " " << rhsRawDTypep->prettyDTypeNameQ()
+                                << " cannot be assigned to " << lhsDTypep->prettyDTypeNameQ());
```

Comment:
"cannot" might not be obvious maybe this?

```suggestion
            nodep->v3error(side << " class " << rhsRawDTypep->prettyDTypeNameQ()
                                << " cannot be assigned to non-class " << lhsDTypep->prettyDTypeNameQ());
```